lib/client.js

```diff
@@ -1412,6 +1460,10 @@ function write (client, request) {
     header += headers
   }
 
+  if (channels.sendHeaders.hasSubscribers) {
+    channels.sendHeaders.publish({ request, headers: header, socket })
```

Comment:
this def will achieve what i mentioned in https://github.com/nodejs/undici/issues/1032 however the headers are a string so it's now on the user to parse that string which to me is not ideal. Is there a reason why the headers aren't associated with the request all the time?

Reply:
I guess you can also get what you need from the socket object so my previous comment/question may be moot

Reply:
> I guess you can also get what you need from the socket object 

Not sure I understood what you are trying to get form the socket

> so it's now on the user to parse that string 

Undici operates headers as string, IMO the parsing should not be done by undici this extra work may affect performance, but the user may not use it at all, the headers string follows the HTTP standard (\r\n separator) should be easy to parse

Reply:
In #1032 I requested outgoing headers, specifically the host header.  Since headers in client are just a string and not key/values pairs like in request it would've been a pain to parse a string to extract the host header.  However, since the socket is here it has the host that it is connected to which is the same so I can just rely on the socket object to get what I need instead of parsing headers string

Reply:
> key/values pairs like in request

If u add any headers to request, please note, on request object they are also string a string 
and should be added as follow `request.addHeader('hello', 'world')`

Reply:
yep, you can see in this prototype I'm using `addHeader` https://github.com/newrelic/node-newrelic/pull/901/files#diff-9fe41f6ad1e32c80844e1948f016e7d692ea907edd2620e8b9245c1c9da67b37R70

Reply:
Here u are using `request.headers` as object, but it is a string 

```
https://github.com/newrelic/node-newrelic/pull/901/files#diff-9fe41f6ad1e32c80844e1948f016e7d692ea907edd2620e8b9245c1c9da67b37R54
```

see this test
https://github.com/nodejs/undici/blob/main/test/diagnostics-channel/get.js#L41

Reply:
yea i noticed that. Honestly this is copied code from our http-outbound and I don't see how a request header could ever be a symbol when serialized